dwds/lib/src/debugging/inspector.dart

```
@@ -101,7 +101,98 @@ class AppInspector extends Domain {
     return inspector;
   }
 
-  Future evaluate(String isolateId, String targetId, String expression,
+  /// Get the value of the field named [fieldName] from [receiver].
+  ///
+  /// Note that the returned [RemoteObject] might be for a simple value, and
+  /// [AppInspector._asDartObject] can be used to get the value.
+  Future<RemoteObject> loadField(RemoteObject receiver, String fieldName) {
+    var load = '''
+        function() {
+          return require("dart_sdk").dart.dloadRepl(this, "$fieldName");
+        }
+        ''';
+    return _callFunctionOn(receiver, load, _marshallArguments([]));
+  }
+
+  /// Call a method by name on [receiver], with arguments [positionalArgs] and [namedArgs].
+  ///
+  /// Note that the returned [RemoteObject] might be for a simple value, and
+  /// [AppInspector._asDartObject] can be used to get the value.
+  Future<RemoteObject> sendMessage(RemoteObject receiver, String methodName,
+      [List positionalArgs = const [], Map namedArgs = const {}]) async {
+    var send = '''
+        function (positional) {
+          if (!(this.__proto__)) { return 'Instance of PlainJavaScriptObject';}
+          return require("dart_sdk").dart.dsendRepl(this, "$methodName", positional);
+        }
+        ''';
+    // TODO(alanknight): Support named arguments.
```

we should throw an `UnsupportedError` if named args are provided

---

Done.